fenixedu-ist-giaf-invoices/src/main/java/pt/ist/fenixedu/giaf/invoices/EventProcessor.java

```diff
@@ -133,7 +135,14 @@ private static void processSap(final ErrorLogConsumer errorLog, final EventLogge
                 } else if (accountingEntry instanceof DebtExemption) {
                     if (accountingEntry.getAmount().compareTo(BigDecimal.ZERO) > 0 && !sapEvent.hasCredit(accountingEntry.getId())
                             && accountingEntry.getCreated().isAfter(EventWrapper.SAP_TRANSACTIONS_THRESHOLD)) {
-                        sapEvent.registerCredit(event, (CreditEntry) accountingEntry, eventWrapper.isGratuity());
+                        final DebtExemption debtExemption = (DebtExemption) accountingEntry;
+                        if (EventExemptionJustificationType.CUSTOM_PAYMENT_PLAN.name().equals(debtExemption.getDescription())) {
```

Comment:
I think this will fail if it doesn't match with localized name.

Reply:
The description of the DebtExemption is "hacked" when it is placed into the calculator:

   https://github.com/Luis-Cruz/fenixedu-academic/blob/b20da390d33c8ecaaf67e1b81822df9299fe93b6/src/main/java/org/fenixedu/academic/domain/accounting/Event.java#L565-L568

so it does not rely on a localized name.

Reply:
what about [here](https://github.com/Luis-Cruz/fenixedu-academic/blob/b20da390d33c8ecaaf67e1b81822df9299fe93b6/src/main/java/org/fenixedu/academic/domain/accounting/Event.java#L1254) ?